pravega/pravega
# Issue 4129: Ensure that a failure to connect propagates upward

Signed-off-by: Tom Kaitchuck <tom.kaitchuck@emc.com>

**Change log description**  
Ensure that a failure to connect propagates upward 

**Purpose of the change**  
Fixes #4129 

**What the code does**  
Inside of `ConnectionPool.establishConnection()` two futures are created, and the output is dependent on `allOf` them. The problem with this is the method explicitly blocks until both complete even in the exception case. So if the first future fails, the returned future will block forever as the second future will never be completed. This changes the logic so that the second future depends on the first.

**How to verify it**  
System tests should be run against this branch to verify the problem does not reoccur.
